### PR TITLE
DR | Add related content to message-aria-describedby

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > - Adds `message-aria-describedby` to add issue page inputs so screen readers include related context
  > - Add description slot to Supplemental Claim's 5103 notice to ensure that related context is read out by a screen reader
- _(If bug, how to reproduce)_
  > In any decision review form, use screen reader on the add issue page the paragraph above the input will not get read out  
- _(What is the solution, why is this the solution)_
  > Use `message-aria-describedby` available in `va-text-input` and description slot available in `va-checkbox`
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- [#63227](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63227)
- [#56080](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56080) (needed component library update)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Related text not read out
- _Describe the steps required to verify your changes are working as expected_
  > Test with screen reader in review instance
- _Describe the tests completed and the results_
  > Manual screen reader testing
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop | Screenshot 2023-09-06 at 10 09 49 AM | PXL_20230906_143330431 |

## What areas of the site does it impact?

Decision review forms (995, 996 & 10182)

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/995/components/AddIssue.jsx [hunk 1]
```diff
@@ -181,6 +181,8 @@ const AddIssue = ({ data, goToPath, setFormData, testingIndex }) => {
           onInput={handlers.onIssueNameChange}
           onBlur={handlers.onInputBlur}
           error={((submitted || inputDirty) && showIssueNameError) || null}
+          message-aria-describedby={content.name.hintText}
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/components/AddIssue.jsx:184]: why do we have two here?
`message-aria-describedby` and `messageAriaDescribedby`

File: src/applications/appeals/995/components/ContestableIssuesWidget.jsx [hunk 2]
```diff
@@ -237,7 +237,7 @@ const ContestableIssuesWidget = props => {
             className="add-new-issue vads-c-action-link--green"
             to={{
               pathname: '/add-issue',
-              search: `?index=${items.length},new=true`,
+              search: `?index=${items.length}`,
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/components/ContestableIssuesWidget.jsx:240]: how come `new=true` is no longer needed?

File: src/applications/appeals/995/components/AddIssue.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/995/components/AddIssue.jsx:184]: LOL I was trying to figure out which one the React binding would accept. It work, so it's one of these two - I'll remove the one that doesn't work. Thanks!

File: src/applications/appeals/10182/components/AddIssue.jsx [hunk 3]
```diff
@@ -185,6 +185,7 @@ const AddIssue = props => {
           onInput={handlers.onIssueNameChange}
           onBlur={handlers.onInputBlur}
           error={((submitted || inputDirty) && showIssueNameError) || null}
+          messageAriaDescribedby={content.hintText}
```
Comment (rocio-desantiago) [inline src/applications/appeals/10182/components/AddIssue.jsx:188]: should this be `content.name.hintText` like in 996?

File: src/applications/appeals/995/components/ContestableIssuesWidget.jsx (hunk 2 above)
Comment (Mottie) [reply src/applications/appeals/995/components/ContestableIssuesWidget.jsx:240]: It wasn't ever needed. I was trying to figure out an easy way to pass data between pages without saving it to the form data or to session storage

**Review by rocio-desantiago — CHANGES_REQUESTED** [review]
A little confused on the different formatting of `content.name.hintText` across the 3

File: src/applications/appeals/10182/components/AddIssue.jsx (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/10182/components/AddIssue.jsx]: oops, yes!

File: src/applications/appeals/995/components/AddIssue.jsx [hunk 4]
```diff
@@ -181,6 +181,8 @@ const AddIssue = ({ data, goToPath, setFormData, testingIndex }) => {
...
+          messageAriaDescribedby={content.name.hintText}
```
[the remark this answers is not in the document]
Comment (rocio-desantiago) [reply src/applications/appeals/995/components/AddIssue.jsx]: lol nvm I don't know why eyes thought it was inside `issueErrorMessages`

**Review by rocio-desantiago — APPROVED** [review]

**Pull request merged**